homeassistant/components/sensorpush_cloud/sensor.py

```diff
@@ -0,0 +1,180 @@
+"""Support for SensorPush Cloud sensors."""
+
+from __future__ import annotations
+
+import itertools
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    ATTR_DEVICE_ID,
+    ATTR_MODEL,
+    ATTR_NAME,
+    ATTR_TEMPERATURE,
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    UnitOfElectricPotential,
+    UnitOfLength,
+    UnitOfPressure,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_BLUETOOTH, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    ATTR_ALTITUDE,
+    ATTR_ATMOSPHERIC_PRESSURE,
+    ATTR_BATTERY_VOLTAGE,
+    ATTR_BLUETOOTH,
+    ATTR_DEWPOINT,
+    ATTR_HUMIDITY,
+    ATTR_LAST_UPDATE,
+    ATTR_SIGNAL_STRENGTH,
+    ATTR_VAPOR_PRESSURE,
+    DOMAIN,
+    LOGGER,
+    MANUFACTURER,
+    MAX_TIME_BETWEEN_UPDATES,
+)
+from .coordinator import SensorPushCloudConfigEntry, SensorPushCloudCoordinator
+
+SENSORS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key=ATTR_HUMIDITY,
+        device_class=SensorDeviceClass.HUMIDITY,
+        translation_key="humidity",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_TEMPERATURE,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        translation_key="temperature",
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+DISABLED_SENSORS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key=ATTR_ALTITUDE,
+        device_class=SensorDeviceClass.DISTANCE,
```

Comment:
I am not sure if you can use .Distance for 'altitude'. The values are correct but it is the same as comparing 'length' and 'distance': they are not the same.

Reply:
I looked through the `SensorDeviceClass` enum and couldn't find anything else that would be applicable. Altitude is the vertical distance from sea level, which seemed like the best fit. Is there a better way to express this?

Reply:
Sensors that do not match deviceclasses are defined without, with the appropriate unit, icon and translation key. But maybe this deviceclass fits, lets wait for another review 😉

Reply:
Distance is fine i think